crypto/asn1/ameth_lib.c

```diff
@@ -216,6 +216,17 @@ EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_new(int id, int flags,
             goto err;
     }
 
+    /*
+     * One of the following must be true:
+     *
+     * pem_str == NULL AND ASN1_PKEY_ALIAS is set
+     * pem_str != NULL AND ASN1_PKEY_ALIAS is clear
+     *
+     * Anything else is an error and may lead to a corrupt ASN1 method table
+     */
+    if ((pem_str == NULL) != ((flags & ASN1_PKEY_ALIAS) == 0))
```

Comment:
... and its negation would yield `(pem_str == NULL) == ((flags & ASN1_PKEY_ALIAS) != 0))`, right?

Reply:
My fault, I overlooked the other `!=`. Your condition is just as correct as mine.

Reply:
Arrggh...!  The condition _is_ wrong! It should be two unequal signs. Or am I completely confused now?

 ```
(pem_str == NULL) != ((flags & ASN1_PKEY_ALIAS) != 0))
 ```

Reply:
Oh!  Quite right.  Fixing